woocommerce/woocommerce
# Migrate merchant daily smoke tests to Playwright

### All Submissions:


### Changes proposed in this Pull Request:

Currently, our E2E tests have a 100% passing rate on PR's where they're run against localhost. However, our E2E tests fail when they're run against our nightly smoke test site due to causes like:

- hardcoded values like login credentials, assertions, and other kinds of test data that are only valid on localhost
- timeouts due to slower performance of our nightly smoke test site compared to a local testing environment
- some pages and menu items in the nightly smoke test site looking a bit differently from localhost
- the fact that our nightly smoke test site is not in a pristine state. It doesn't have the default settings anymore, and it most often contains leftover products, orders, and other items from previous test runs

This pull request therefore aims to make our E2E tests pass on the nightly smoke test site. Additionally, it aims to run the E2E tests on `trunk` with certain plugins just like what we used to do on the Puppeteer tests previously.

Note to PR reviewer:

1. I placed PR comments on certain parts of the code to hopefully help with the review.
1. This PR runs all E2E tests except for the `shopper` ones. Shopper tests will be addressed on a separate PR soon.
1. The job `API tests on nightly build` currently passes just because only the `hello.test.js` test was being run.

Closes https://github.com/woocommerce/woocommerce-quality/issues/418.

### How to test the changes in this Pull Request:

1. Navigate to the [Smoke test daily](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml) workflow runs page.
2. In the "Run workflow" dropdown, select this branch (`e2e/daily-playwright-2`), then click "Run workflow".
3. Refresh the page, and click on the in-progress "Smoke test daily" run.
4. Wait for the workflow to finish. You should see all tests pass like so:
   <img width="1004" alt="ZwAQJbELXU" src="https://user-images.githubusercontent.com/4509348/200018533-dbaba51c-38ac-487e-a044-ed0dd22d707f.png">
5. Rerun the workflow if it had any failures. Occasionally, a few tests fail due to very weird reasons like:
   - the login page doesn't load at all for some reason despite the retries
   - `E2E_GH_TOKEN` wasn't being accepted for some reason
6. Scroll down the summary page of the workflow run. You should see:
   - Test report artifacts
   - "Smoke tests on trunk" job summary
      <img width="661" alt="aaHepaR2Pi" src="https://user-images.githubusercontent.com/4509348/200018623-6bd06b2a-b344-42be-b22c-2e48137def9d.png">

7. In the "Smoke tests on trunk" job summary, click on each of the links at the footer to verify if you're correctly taken to the API and E2E Allure reports, and to the WooCommerce Test Reports Dashboard.
8. Go to [`#woo-test-reports-daily`](https://a8c.slack.com/archives/C03H038EV2R) and notice that the E2E test stats are not TBC anymore. They now show actual numbers:
   <img width="463" alt="rKVYvefjYX" src="https://user-images.githubusercontent.com/4509348/200018755-6d78f748-dcc3-4015-af9f-a6038ba4a38a.png">

9. Click on the 'View full E2E report' button. You should be taken to the "Daily E2E Smoke Test Report".

### Other information:




### FOR PR REVIEWER ONLY:
